### ISSUE TITLE
[BUG] Cannot add multiple attribute decorators for same property

### ISSUE THREAD

**Issue by lnovelli** [blazejkustra/dynamode#30]
### Summary:
Hi @blazejkustra ! BIG kudos for developing this library, I truly enjoy using it.

I'm bumping into an issue that I think is related to the recently closed #27 . The bug appears when I try to add 2 attribute decorators (one main attribute, one GSI attribute) to a single property.  This is normally possible when using CDK or the AWS console.

### Code sample:

```typescript
export class TableUser extends Entity {

  @attribute.partitionKey.string()
  @attribute.gsi.partitionKey.string({ indexName: 'index-1' })
  domain: string;

  @attribute.sortKey.string()
  @attribute.gsi.sortKey.string({ indexName: 'index-1' })
  email: string;
  
}
```

My TableManager is defined as follows:
```typescript
export const UserdataTableManager = new TableManager(TableUser, {
  tableName: USERS_TABLE_NAME,
  partitionKey: 'domain',
  sortKey: 'email',
  indexes: {
    'index-1': {
      partitionKey: 'domain',
      sortKey: 'email',
    },
  },
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
});
```

### Action performed:

If I remove the secondary attribute decorators, the code compiles.
Is there anything I'm doing wrong? Let me know if you need any other information to help debug it!

### Expected result:

I'd expect to be able to add multiple attribute decorators to a single property, no matter if they belong to GSI, LSI or main ones.
Unfortunately this is blocking me from using the GSIs, but I LOVE this package and I hope we can find a fix.

### Actual result:

I get this compile error:

`DynamodeError [DynamodeStorageError]: Attribute "domain" was already decorated in entity "TableUser"`

### Environment:

MacOS Sonoma 14.4.1
Node v20.14

### Other:
- [ X] I have read through the Dynamode documentation before posting this issue
- [ X] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this issue has not already been raised before
- [ X] I have searched the internet to ensure this issue hasn't been raised or answered before

**Comment by blazejkustra** [comment]
Hey, thank you! 🙌 Tbh I haven't thought about a case where primary key would be also a GSI at the same time.

Can you describe what would be the purpose for such index?

**Comment by lnovelli** [comment]
Yes, I know it might sound a bit strange. I have a DB where some attributes are VERY heavy, so I'm creating a secondary index where I only project the lighter attributes, but I keep the same access modality (same partition/sort key). Does it make sense?

**Comment by blazejkustra** [comment]
Ah yes, I forgot about this use case, it makes sense. Thank you!

**Comment by afronski** [comment]
@blazejkustra This is also the case I missed here https://github.com/build-on-aws/aws-lambda-and-cqrs-a-winning-combination/commit/17439a4f3d456f021f2d13b58be49331ea426094#r136008382 and we discussed it briefly.

Long story short: being able to add multiple decorators to the same property allows for a technique called _index overloading_, where you utilise an existing attribute multiple times. It is usually used to save on WCU (it may sound like penny pinching, but in case of tables with a lot of writes it adds up) and for easier way to preserve consistency without redundant values across multiple attributes.

**Comment by blazejkustra** [comment]
I did a little research and know how to code it but didn't have the time last week. Thank you for your proposition @afronski, any contributions are welcome! In this case I would also be glad if you could review the changes once I create a PR, wdyt?

**Comment by afronski** [comment]
@blazejkustra sounds good to me - happy to help, feel free to ping me when it's ready!

**Comment by blazejkustra** [comment]
https://github.com/blazejkustra/dynamode/pull/31 is up @afronski, please have a look!

**Comment by afronski** [comment]
@blazejkustra thank you so much! I will post my remarks in #31.

**Comment by blazejkustra** [comment]
https://github.com/blazejkustra/dynamode/pull/31 was just merged 😄 The issue should be fixed in v1.4.0! Let me know if it works as you expected @lnovelli. Thanks for your help again @afronski

**Comment by lnovelli** [comment]
It's perfect! Many thanks @blazejkustra and @afronski 🙌
